src/Tests/Units/Client/Client.php

```diff
@@ -77,6 +77,40 @@ public function testHandleEventWithImmediateSend()
     }
 
     /**
+     * Test handleEvent with to send limit
+     */
+    public function testHandleEventWithToSendLimit()
+    {
+        $client = $this->getMockedClient();
+        $client->setToSendLimit(3);
+
+        $event = new \Symfony\Component\EventDispatcher\Event();
+
+        $queue = new \SPLQueue;
+
+        $client->getMockController()->increment = function($value) use ($queue) {
+            $queue->enqueue($value);
+        };
+        $client->getMockController()->getToSend = $queue;
+
+        $client->addEventToListen('test', array(
+            'increment' => 'stats.<name>',
+        ));
+
+        $this
+            ->if($client->handleEvent($event, 'test'))
+            ->then
+                ->mock($client)
+                    ->call('send')->never()
+            ->then($client->handleEvent($event, 'test'))
+                ->mock($client)
+                    ->call('send')->never()
+            ->then($client->handleEvent($event, 'test'))
+                ->mock($client)
+                    ->call('send')->once();
+    }
```

Comment:
I would add another 3 calls to `handleEvent()` after these to check that it works more than one time.